### PR TITLE
[docs] APM Overview structure and content improvements

### PR DESCRIPTION
**Pull request by bmorelli25**
**Preview this PR on Firebase:** https://elastic-obs-docs.firebaseapp.com/index.html
😮 🆒 😎 🎉 

---

This PR improves the content and structure of the APM Overview:
* Shifts to a cloud-first approach for installation
* Improves the layout of tables
* Refreshes old images and screenshots
* Adds new architecture images - closes https://github.com/elastic/apm-server/issues/2117
* Adds a new "Troubleshooting" top level heading
* Adds a new "Real User Monitoring" top level heading
* Updates the overview
* Updates the components and documentation links

### Proposed new structure:

* **Overview** (What is APM?)
* **Components and documentation** (What are the components and where can I find the relevant documentation?)
* **Install and run** (Cloud first)
  * Step 1: Install Elasticsearch
  * Step 2: Install Kibana
  * Step 3: Install APM Server
  * Step 4: Install APM agents
* **Data Model** (What kind of data does APM capture?)
  * Spans
  * Transactions
  * Errors
  * Metrics
  * Metadata
* **Distributed tracing**
* **Real User Monitoring (RUM)**
* **OpenTracing bridge**
* **Agent and Server compatibility**
* **Troubleshooting**
* **Breaking changes**
* **Release highlights**

Other notes:

* These changes are similar to the changes proposed for APM Server in (WIP) https://github.com/elastic/apm-server/pull/2170.
* I left out a "getting started" section until we're able to define exactly what that is.

### CODE REVIEW

**Comment by bmorelli25** [comment]
The failing elasticsearch-ci/docs-build is unrelated (it's really a bug).

**Comment by bmorelli25** [comment]
cc @tbragin

File: docs/guide/troubleshooting.asciidoc [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+* {apm-server-ref-v}/troubleshooting.html[Server troubleshooting]
+* {apm-java-ref-v}/trouble-shooting.html[Java agent troubleshooting]
+* {apm-node-ref-v}/troubleshooting.html[Node.js agent troubleshooting]
+* {apm-rum-ref-v}/troubleshooting.html[RUM troubleshooting]
+
+[float]
+=== GitHub repositories
+
+Elastic APM is open source! Our https://github.com/elastic?utf8=%E2%9C%93&q=apm[GitHub repositories] have a plethora of knowledge available in the form of code, issues, and PRs.
```
Comment (graphaelli) [inline docs/guide/troubleshooting.asciidoc]: GH repos are typically for "verified bug reports, feature requests, and pull requests." - can we break this out from the troubleshooting section?

File: docs/guide/troubleshooting.asciidoc [hunk 2]
```diff
@@ -0,0 +1,25 @@
+[[troubleshooting]]
+== Troubleshooting
+
+If you run into trouble, there are three places you can look for help.
```
Comment (graphaelli) [inline docs/guide/troubleshooting.asciidoc]: How about paid support?

Comment (bmorelli25) [reply docs/guide/troubleshooting.asciidoc]: 👍Swapped in place of GitHub

**Comment by tbragin** [comment]
@bmorelli25 Love this change! I see it in master:
https://www.elastic.co/guide/en/apm/get-started/master/install-and-run.html#install-and-run

How far back do you think we can backport it?

**Comment by tbragin** [comment]
Small nit -- since the page above is so short, I don't know if we need the "Alternatively, you can install the stack yourself." since the install the stack option is visible above the fold.

**Comment by bmorelli25** [comment]
Thanks Tanya. Good point with the nit, I'll get that fixed.

Backporting to `7.0` will be easy. Everything before that will take a little bit of work. I'm going to try and take this as far as I can. Probably `6.6` or `6.5`.

**Pull request merged**